syntax/zeek.vim

```diff
@@ -2,13 +2,21 @@
 " Language: Zeek (https://zeek.org)
 " Author: Jon Siwek
 
-if exists("b:current_syntax")
+if exists('b:current_syntax')
   finish
 endif
 
+" For highlighting payloads to `@TEST-EXEC*`.
+syntax include @zeekSh syntax/sh.vim
+
 syn match zeekComment /#.*/ contains=zeekTodo
 syn keyword zeekTodo TODO XXX FIXME NOTE contained
 
+syntax match zeekBTest /\v\@TEST(-\w+)+:?.*/ containedin=zeekComment containedin=ALL " This group only resets syntax.
+syntax region zeekBTestExec start=/@TEST-\(EXEC\|REQUIRES\).\{-}\s/ end=/$/ containedin=zeekBTest contains=@zeekSh
+syntax match zeekBTestKeyword /@TEST-.\{-}\s/ containedin=zeekBTestExec containedin=zeekBTestOther
+syntax match zeekBTestKeyword /@TEST-END-FILE/ containedin=zeekBTest " Extra case for keywords which do not take args.
+
```

Comment:
Hmm, something's not quite right or is it expected that TEST-START-FILE TEST-DOC and TEST-GROUP TEST-IGNORE don't function, or is that on my end?

![Screenshot-20230904-100835](https://github.com/zeek/vim-zeek/assets/2100509/bdd44e71-1c3c-43fe-b915-a5f6dca9e097)

(using solarized, colorscheme=dark)

Reply:
Yepp, they were missing. I added highlighting for them in a fixup commit.

Reply:
Ah, there's more, but okay to fix'em up as they come I guess?

TEST-PORT and TEST-START-NEXT are definitely used often, so maybe those before merging?

TEST-MEASURE-TIME and TEST-KNOWN-FAILURE exist while skimming the docs.

There's also the situation that the `@TEST-` part is configurable, but I guess this isn't a highlighter for btest, it's a highligher for Zeek scripts that includes btest snippets and in that case assuming `@TEST` seems fine.

Reply:
I added all the ones you pointed out, but we can always add more in case we are annoyed by it. I would have preferred to avoid enumerating BTest keywords by hand, but the way I set up the highlighter requires this. This works, but I am definitely open to suggestions how to improve it.